### PR TITLE
Adding new cropByFocalArea filter

### LINKED ISSUE #121: Implement new crop by focus area filter
A filter that based on a focal point and desired width and height returns the cropped version of a give image

### PR DESCRIPTION
**Pull request by githubdoramon**
Adding new filter that generates cropped image based on desired dimensions on centered focal point.

Fixes #121

### CODE REVIEW

File: filters/cropbyfocalarea.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	return &bimg.Options{
+		AreaWidth:  desired_width,
+		AreaHeight: desired_height,
+		Top:    focalPointY - desired_height/2,
+		Left:    focalPointX - desired_width/2,
+		}, nil
+}
+
+func (f *cropByFocalArea) CanBeMerged(other *bimg.Options, self *bimg.Options) bool {
+	return (other.AreaWidth == 0 || other.AreaWidth == self.AreaWidth) &&
```
Comment (danpersa) [inline filters/cropbyfocalarea.go]: Why not returning false? Can you explain the logic?

```suggestion
	return false
```

Comment (githubdoramon) [reply filters/cropbyfocalarea.go]: The idea was to allow the filter to be merged in any other case that is not dealing with the cropping specifically. But if you think is better to never allow merge, for sure we can go in this direction

File: filters/cropbyfocalarea.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			(other.AreaHeight == 0 || other.AreaHeight == self.AreaHeight) &&
+			(other.Top == 0 || other.Top == self.Top) &&
+			(other.Left == 0 || other.Left == self.Left) &&
+			(other.Width == 0) &&
+			(other.Height == 0)
+}
+
+func (f *cropByFocalArea) Merge(other *bimg.Options, self *bimg.Options) *bimg.Options {
```
Comment (danpersa) [inline filters/cropbyfocalarea.go]: If the previous function returns false, this one should return null

File: filters/cropbyfocalarea.go (hunk 1 above)
Comment (danpersa) [reply filters/cropbyfocalarea.go]: I would not for not merging at all, as applying cropping two times is error prone and should probably never happen.

**Comment by rgritti** [comment]
:+1:

**Comment by danpersa** [comment]
Will be added to Walk of Fame, the Zalando internal version of skrop

**Pull request closed without merging**